src/main/java/org/alliancegenome/curation_api/services/ConstructService.java

```diff
@@ -119,8 +123,13 @@ public Construct removeOrDeprecateNonUpdated(Long id, Boolean throwApiError, Str
 		}
 		
 		Boolean anyReferencingEntities = false;
-		// TODO: implement check for any referencing entities and code for their deprecation
-		// once links between constructs and other entities have been established
+		if (CollectionUtils.isNotEmpty(construct.getConstructGenomicEntityAssociations())) {
```

Comment:
This piece of code is duplicated in multiple places. Could that be re-used in a single class?

Reply:
There are class-dependent differences

Reply:
Sure, will take a look at it later and put in a separate PR so we can get this stuff in and the associations loaded before sprint review.